components/aws/sagemaker/common/_utils.py

```diff
@@ -254,6 +254,13 @@ def get_image_from_job(client, job_name):
     return image
 
 
+def stop_training_job(client, job_name):
```

Comment:
All these functions seem redundant 
all of them are basically calling the stop function from boto3 client and catching the exception 

Instead of individual functions for each, we can have 1 function with type_of_job argument 
Eg : 
```
def stop_sagemaker_job(client, type_of_job, job_name):
    try:
        if type_of_job=="training":
            client.stop_training_job(TrainingJobName=job_name)
        elif type_of_job=="transform":
            client.stop_transform_job(TransformJobName=job_name)
          .
          .
          .
          .
    except:

```

Reply:
I think this would be better suited to inheritance structures, but I agree duplicated code like this becomes annoying.

Reply:
Agree about duplication, but its better to keep it separated since each is being used by different component like create/describe request

Ideally we should have utils per component